crates/bevy_ui/src/render/mod.rs

```diff
@@ -173,29 +184,81 @@ pub fn extract_uinodes(
             Option<&UiImage>,
             &ComputedVisibility,
             Option<&CalculatedClip>,
+            Option<&Handle<TextureAtlas>>,
```

Comment:
I used the option here because that was the same style used for UI image, otherwise I would have moved this to a completely seperate extract function.  My concern is that with multiple queries we would match the same image (or imageless node) multiple times but there might be a way around this.

Reply:
Yep, just add a `Without<UiImage>` query filter to the texture atlas query and it should be fine.